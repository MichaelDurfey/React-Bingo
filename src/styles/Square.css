/* @import url('https://fonts.googleapis.com/css?family=Ubuntu'); */

:local(.square) {
  display: flex;
  align-items: center;
  justify-content: center;
  border-left: solid #dfdfdf 1px;
  border-top: solid #dfdfdf 1px;
  font-family: 'Ubuntu', sans-serif;
  color: #90a2a7;
  font-weight: 100;
  font-size: 13px;
  background-color: #fff;
}

:local(.square:nth-child(5n)) {
  border-right: solid #dfdfdf 1px;
}

:local(.square:nth-last-child(-n+5)) {
  border-bottom: solid #dfdfdf 1px;
}

:local(.squareSelected) {
  background-color: #0070BA;
  color: white;
}