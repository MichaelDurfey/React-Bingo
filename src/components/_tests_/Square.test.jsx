import React from 'react';
import renderer from 'react-test-renderer';
import Square from '../Square';

describe('Square', () => {
  test('the component renders with correct classNames if Selected', () => {
    const square = renderer
      .create(<Square selected={false} number={63} />)
      .toJSON();
    // console.log(wrapper.html())
    expect(square).toMatchSnapshot();
  });
  test('the component renders with correct classNames if not Selected', () => {
    const square = renderer
      .create(<Square selected={false} number={25} />)
      .toJSON();
    expect(square).toMatchSnapshot();
  });
});
