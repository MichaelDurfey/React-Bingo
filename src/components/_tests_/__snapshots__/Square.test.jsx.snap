// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square the component renders with correct classNames if Selected 1`] = `
<div
  className="square"
  role="presentation"
>
  63
</div>
`;

exports[`Square the component renders with correct classNames if not Selected 1`] = `
<div
  className="square"
  role="presentation"
>
  25
</div>
`;
